File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int count = 0;
        for(int i =0;i<nums.size();i++){
            if(nums[i]==0){
                count++;
            }
        }
        if(count>0){
            sort(nums.begin(),nums.end());
        if(nums[0]==0 && (nums[nums.size()-1]==0)) return;
        while(nums[0]==0 && nums.size() > 1){
            nums.push_back(nums[0]);
            nums.erase(nums.begin());
        }
            

        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        /**fuck what is the logic
        unless anything other one zero put it forward then add zero
         */
         int j =0;
         int count= 0;
         for(int i =0;i<nums.length;i++){
            if (nums[i]==0){
                count++;
            }
            if(nums[i]!=0){
                
                nums[j]=nums[i];
                j++;
            }
            
         }
         int n = nums.length-1;
         for(int i =0;i<count;i++){
            nums[n-i]=0;
         }
    }
}